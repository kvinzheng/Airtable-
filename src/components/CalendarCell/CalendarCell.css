.CalendarCell-day {
  text-align: center;
  font-size: 0.9rem;
  padding: 2px 3px 0 0;
}

.CalendarCell {
    background: white;
    border-bottom: 1px solid #dfdfdf;
    border-right: 1px solid #dfdfdf;
    min-height: 8rem;
    display: flex;
    flex-direction: column;
    height: 100vh;
}
  