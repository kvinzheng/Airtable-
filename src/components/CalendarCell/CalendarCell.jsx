import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import CalendarEventList from "../CalendarEventList/CalendarEventList";
import "./CalendarCell.css";

const CalendarCell = ({
  date,
  cellIndex,
  type,
  color,
  eventPosition,
  lastDate,
  zoomWidth,
}) => {
  return (
    <div
      className={"CalendarCell " + type}
      style={{ width: `${zoomWidth}px` }}
    >
      <div className="CalendarCell-day">{date.getDate()}</div>
      <CalendarEventList
        date={date}
        color={color}
        cellIndex={cellIndex}
        eventPosition={eventPosition}
        lastDate={lastDate}
      />
    </div>
  );
};

export const mapStateToProps = (state) => {
  return {
    zoomWidth: state.events.zoom.width,
  };
};

CalendarCell.propTypes = {
  date: PropTypes.object,
  cellIndex: PropTypes.number,
  type: PropTypes.string,
  color: PropTypes.string,
  eventPosition: PropTypes.array,
  lastDate: PropTypes.object,
  zoomWidth: PropTypes.number,
};

export default connect(mapStateToProps, {})(CalendarCell);
