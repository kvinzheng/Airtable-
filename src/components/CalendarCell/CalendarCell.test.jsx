import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";

import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureMockStore from "redux-mock-store";

import CalendarCell from "./CalendarCell";

configure({ adapter: new Adapter() });
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

describe("CalendarCell", () => {
  it("should render without any data defined", () => {
    const component = shallow(
      <Provider store={store}>
        <CalendarCell zoomWidth={150} />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
  
});
