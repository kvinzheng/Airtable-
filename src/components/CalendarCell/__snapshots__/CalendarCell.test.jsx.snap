// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarCell should render without any data defined 1`] = `ShallowWrapper {}`;
